TicketsDemo.Domain/DefaultImplementations/HolidayPriceCalculationStrategy.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    public class HolidayPriceCalculationStrategy : IPriceCalculationStrategy
+    {
+        private IPriceCalculationStrategy _calculationStrategy;
+        private IHolidayRepository _holidayRepository;
+        public HolidayPriceCalculationStrategy(IPriceCalculationStrategy strategy, IHolidayRepository holidayRepository)
+        {
+            _calculationStrategy = strategy;
+            _holidayRepository = holidayRepository;
+        }
+        public List<PriceComponent> CalculatePrice(PlaceInRun placeInRun)
+        {
+            List<PriceComponent> components = new List<PriceComponent>();
+            Holiday holiday = _holidayRepository.GetHolidayDetails(DateTime.Now);
+            decimal weekendPercent = Weekend(DateTime.Now);
```

Comment:
please move this line to be located right above the `if (weekendPercent != 0)` check.

Now, single logical activity is split thorough the code of method, which makes code harder to read and understand